### ISSUE TITLE
Collection of ideas and improvements for aux handling

### ISSUE THREAD

**Issue by SammysHP** [ToyKeeper/anduril#20]
For my own purposes I had refactored some aux related code. This includes:

- (previously an `indicator_led_update()` function, which is now already included upstream)
- refactoring of aux LED config to support more blinking patterns instead of only four
- more blinking patterns: simple low blink, simple high blink
- disable aux LEDs, but keep on button LED
- some improvements when aux LEDs are turned on (@ToyKeeper we shortly talked about it and you said that some of those were intentional)

https://github.com/SammysHP/flashlight-firmware/compare/multi-channel...mc-more-aux-patterns

Maybe you find some helpful modifications in this collection. If you have some roadmap in your mind, I could refactor some of my changes and make them mergeable.

**Comment by lukestein** [comment]
Just a user endorsement of the value of a low blink. This allows having two low-only aux modes (steady low and low blink) to distinguish off from lockout on lights without rgb aux while protecting dark-adapted vision and battery life. Would love to see this in the main builds 🙏

**Comment by SiteRelEnby** [comment]
FWIW I specifically like the idea of a high blink, or also constant-on low with the occasional high blink, which I think would be the best for locating a light in the dark, with the blink for if the aux are facing away so harder to see on low (but without them spending time completely off, also for some reason I find the current high/low blink animation stressful to watch, although can't really explain why). Plus, low with high blink would be great on disco mode.

**Comment by SammysHP** [comment]
Indeed. I've added it to my fork as well, but with so many modes 7C gets very annoying. I think we need a menu to configure aux LEDs. Something like 7H enters the menu, items are mode, pattern and maybe more aux related settings.

**Comment by ToyKeeper** [comment]
It has been a very slow process, but some of this is implemented in trunk now.

- Done: an indicator_led_update() function, which is now already included upstream
- Done: refactoring of aux LED config to support more blinking patterns instead of only four
- Partial: more blinking patterns: simple low blink, simple high blink
- Done: disable aux LEDs, but keep on button LED
- Unsure: some improvements when aux LEDs are turned on (@ToyKeeper we shortly talked about it and you said that some of those were intentional)

The aux blink patterns are now an array which can have up to 255 total "frames" of animation and metadata.  It now includes 3 blinking patterns, but adding more is easy.  It's just a single array of bytes: `[ anim1_length, anim1_frames..., anim2_length, anim2_frames..., ...]`

The main question is which ones should be included by default.  Currently it has:

- off/low/high (steady, not animated)
- 2-level pattern: low, pause, low-high-low, pause
- 2 low blinks
- 1 low blink

On older lights with just a single-color aux LED, it still uses the previous modes.  Off/low/high, of course, but only one blinky -- the 2-level pattern.  This is mostly to save space.
